### ISSUE TITLE
Moved mailman archives up until Feb 12, 2020 to test1.webwork.maa.org/mailman_archives/public

### ISSUE THREAD

**Issue by mgage** [openwebwork/webwork.maa.org#18]
Archives up to Feb 12 are preserved.  Test to make sure they are secure

**Comment by taniwallach** [comment]
So far, I do not see anything available via http on the new server.

---
Correction: I found the link: https://test1.webwork.maa.org/mailman_archives/archives/public/ in a different issue.
---

It seems that the MailMan archives such as
1. http://webwork.maa.org/pipermail/webwork-devel
2. http://webwork.maa.org/pipermail/webwork-bugs

are actually stored under `archives/public/LISTNAME/` somewhere under "/var" (probably under   `/var/lib/mailman`).

Probably we should transfer a full copy of the old MailMan tree to a safe private place on the new server, and just put a static copy of the `archives/public` as a `pipermail` subdirectory of /var/www/html (or wherever the WWW document root directory was moved to) on the new server. I think that should make the archives available, The use of `pipermail` is to match the expected location, which is where the archives are now served from by Apache.

I think it might make sense to first have mailman (on the old server) change the links to use `https`, and then recreate the archives (see the 3rd reference below), but only after saving away a copy of the current archives on the new server. 

If we want the archive to work now with the test1.webwork.maa.org address, a temporary version of the archive would need to be produced using that hostname instead of the webwork.maa.org hostname, which is probably possible using the same ideas from the 3rd reference below.

Note: It also probably makes sense to make sure that we have convenient long-term access to the "mbox" file MailMan made of the archives we are keeping public, as that seems to be the key to any future work we might want to do with these lists. These files are apparently under a path of the form `archives/private/LISTNAME.mbox/LISTNAME.mbox`. Access to those files is more critical than to the rest of the material we may be storing away.

Note: Since we are closing down the lists, we do not need to replace the signup / info pages like:
  - http://webwork.maa.org/mailman/listinfo/webwork-devel
  - http://webwork.maa.org/mailman/listinfo/webwork-bugs
with anything similar on the new server.

References: 
1. https://mail.python.org/pipermail/mailman-users/2015-February/078487.html
    - (explains about the archives)
2. https://mail.python.org/pipermail/mailman-users/2012-April/073225.html
    -  (explains about the archives)
3. https://mail.python.org/pipermail/mailman-users/2005-December/048001.html
    - (how to rebuild the archive to use https, may be useful)

**Comment by mgage** [comment]
What I have now on  https://test1.webwork.maa.org <https://test1.webwork.maa.org/> is a static copy of what is usually in the pipermail directory.
I planned to rearrange the front page and the links so that the address was more straightforward.

Making a complete (hidden) copy of the mailman directory is probably a good idea.

I’ll get you credentials for webwork.maa.org <http://webwork.maa.org/> and test1.webwork.maa.org <http://test1.webwork.maa.org/> in just a minute.

Take care,

Mike

**Comment by taniwallach** [comment]
If we can run the old mailman, it can handle modifying the URLs being used (see above), although the method you used certainly worked.

**Issue:**

1. Each "main" page of the archive has a line "You can get more information about this list" most of which is a link to a mailman generated page (ex. [for webwork-devel](http://webwork.maa.org/mailman/listinfo/webwork-devel) ) which we are not preserving. 
2. Similar links "More info on this list..." appear in the HTML pages.

We probably want to either remove all such links (lots of work) or create a static page at the relevant location (ex. `/var/www/html/mailman/listinfo/webwork-devel/index.html`) with suitable text that there is no longer an active list, just the archive at X.

**Comment by mgage** [comment]
> On Feb 17, 2020, at 3:57 PM, Nathan Wallach <notifications@github.com> wrote:
> 
> If we can run the old mailman, it can handle modifying the URLs being used (see above), although the method you used certainly worked.
> 
> Issue:
> 
> Each "main" page of the archive has a line "You can get more information about this list" most of which is a link to a mailman generated page (ex. for webwork-devel <https://urldefense.proofpoint.com/v2/url?u=http-3A__webwork.maa.org_mailman_listinfo_webwork-2Ddevel&d=DwMCaQ&c=kbmfwr1Yojg42sGEpaQh5ofMHBeTl9EI2eaqQZhHbOU&r=C6Pt5AGtImanmAdcooarL-JZO8M5dSFPfs3VweYXYkE&m=ofT9s95ZmM48g7gCdtsLUCQP7ql11_kM0mVlZ2REJxw&s=MVh0LJA8wD6J0H_AHhJbb9UiFo5anxBz6Nrz81JkSfA&e=> ) which we are not preserving.
> Similar links "More info on this list..." appear in the HTML pages.


> We probably want to either remove all such links (lots of work) or create a static page at the relevant location (ex. /var/www/html/mailman/listinfo/webwork-devel/index.html) with suitable text that there is no longer an active list, just the archive at X.
> 
> 
I think we can handle this by putting up a page at the appropriate place that points either to google groups and/or to a forum on our moodle.
Probably the latter at this point.

For the ww-bugs list serve I think we should keep using bugzilla at for the PG bugs (which is most of them these days) and people can subscribe
to bugzilla to get on the mailing list.  Bugzilla is mostly working correctly as far as I can tell. 

For other bugs we’ll go to issues —most of the other bugs are being posted either in issues or on the forum already.

Sound ok?

I do think we should mothball all of mailman, so that we can reanimate if we need to.  
I have copies of much of the material on a disk in my homeoffice but it’s a bit out of date since I made it in January when I started all of this.

Take care,

Mike
